### ISSUE TITLE
Address space collision with docker0 bridge

### ISSUE THREAD

**Issue by sairez** [kubernetes/kops#710]
As per a conversation with @justinsb in #kubernetes-users slack:

One of our customers is currently experiencing difficulties accessing nodes and masters via SSH because the CIDR of their client VPN (172.17.0.0/16) overlaps with the CIDR of the docker bridge `docker0`.

In other Docker implementations, we could potentially solve this conflict by adding the bridge IP `--bip` flag to `DOCKER_OPTS` to supply a specific IP address and netmask for the `docker0` bridge.

I'm not sure the best way to obtain this same solution with kops.  It seems as if there is groundwork laid to allow for some customization of certain docker configuration options [here](https://github.com/kubernetes/kops/blob/d3f69496e410630c219fa8ea35032e5a22eb6991/pkg/apis/kops/dockerconfig.go#L23-L35). Is it possible to make `-bip` configurable here and use a default less likely to interfere with common VPN/office/datacenter CIDRs?

**Comment by chrislovecnm** [comment]
@sairez need more details.  Version of K8s, network config in kops, etc.  

I am not sure if "In other Docker implementations, we could potentially solve this conflict by adding the bridge IP --bip flag to DOCKER_OPTS to supply a specific IP address and netmask for the docker0 bridge" is recommended for K8s.  That is what we would need to determine.  I am thinking that it is not, but may be wrong.

**Comment by sairez** [comment]
Sure thing. Here are the bits I think are relevant:

```
$ kops version
Version git-36df075
```

```
  cloudProvider: aws
```

```
  kubernetesVersion: v1.4.3
```

```
networkCIDR: 10.y.0.0/16
  networkID: <redacted>
  networking:
    kubenet: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  zones:
  - cidr: 10.y.8.0/24
    name: us-east-1b
  - cidr: 10.y.9.0/24
    name: us-east-1c
  - cidr: 10.y.10.0/24
    name: us-east-1d
```

Kubernetes cluster is inside a VPC with CIDR `10.y.x.x/16`. 

Masters and nodes reside in private subnets `10.y.8.0/24`, `10.y.9.0/24`, `10.y.10.0/24` across 3 availability zones.

All three private subnets share the same route table (K8s limitation) and thus the same NAT gateway.

OpenVPN appliance exists in public subnet attached to a route table using the IGW as default route.  Clients connect to OpenVPN appliance and are assigned a client IP address out of a pool.  Current VPN client address pool is `172.17.0.0/16`.  

A client's VPN address is routed (not NAT'd) inside the VPC, and so when interacting with internal resources, client retains `172.17.x.x` identity.

**Current Behavior:**

- Clients have success SSHing in to EC2 instances (10.y.x.x) which do not have a docker bridge occupying `172.17.0.0/16`. 

- Clients are unable to SSH into K8s masters and nodes with docker bridge occupying `172.17.0.0/16`

- A user is able to SSH in to EC2 instances (10.y.x.x) and from there then SSH successfully into k8s masters and nodes.

Let me know if there is any further information you'd find useful.

Thanks!

**Comment by chrislovecnm** [comment]
@justinsb you have any recommendations?

**Comment by justinsb** [comment]
Sorry this took so long @sairez   This will be fixed in alpha4 (hopefully today).  You'll be able to pass bip using "expert mode" e.g. https://github.com/kubernetes/kops/pull/1357#issuecomment-271696615

But, TBH, I'm not entirely sure where the docker bridge is used.  Pods should be on the pod network, which is by default 100.64.0.0.  Host network pods should be on the node ips.  So I'm not sure where the conflict arises?

I think this issue will get closed automatically if/when #1574 merges, so maybe reopen (or comment and ask me to reopen) with more details?
